.github/workflows/pages-deploy.yml.hook

```diff
@@ -54,17 +54,14 @@ jobs:
           bundle exec jekyll b
 
           if [[ -n $SPEC_TEST ]]; then
-            # Bypass the defects of htmlproofer
+            # build in a subdir (_site/baseurl) so it will pass the
+            # htmlproofer test
             bundle exec jekyll b -b "" -d "$SPEC_TEST"
           fi
 
       - name: Test Site
         run: |
-          if [[ -n $SPEC_TEST ]]; then
-            bash tools/test.sh -d "$SPEC_TEST"
-          else
-            bash tools/test.sh
-          fi
+          bash tools/test.sh
 
       - name: Deploy
         run: |
```

Comment:
3. At last, we should ensure the site files are placed in the root of `_site` on CI:

```diff
+        if [[ -n $BASE_URL ]]; then
+           mv _site$BASE_URL _site-rename
+           rm -rf _site
+           mv _site-rename _site
+        fi
+          
          bash tools/deploy.sh
```

Reply:
You have one space too much of indentation ;-)